tests/tt_metal/tt_metal/unit_tests_common/CMakeLists.txt

```diff
@@ -28,7 +28,8 @@ set(UNIT_TESTS_COMMON_SRC
     ${CMAKE_CURRENT_SOURCE_DIR}/watcher/test_waypoint.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/watcher/test_link_training.cpp
 )
-add_library(unit_tests_common_o OBJECT ${UNIT_TESTS_COMMON_SRC})
+add_library(unit_tests_common_o STATIC ${UNIT_TESTS_COMMON_SRC})
```

Comment:
Does something not work with `OBJECT` libs here?

Reply:
Yes, Unity build didn't work with OBJECT. I didn't see a slowdown related to changing this OBJECT to STATIC

Reply:
`OBJECT` vs `STATIC` negligible.  I'm just surprised it doesn't work with `OBJECT` given that it's compile-only.  But likely it just wasn't implemented.  `OBJECT` was somewhat of a second-class citizen until later version of CMake.

Reply:
Is this something we have to worry about? Is `3.20` sufficiently "later"?

Reply:
`v3.16` was already "later" enough that we can link against an Object lib directly rather than using the `$<TARGET_OBJECTS>` genex.  I don't know off-hand what other areas Object libs lagged behind Static libs.  But the performance should be so negligible I wouldn't worry about it anytime soon.